contracts/src/v0.8/dev/OptimismValidator.sol

```diff
@@ -1,71 +1,185 @@
+    s_gasLimit = gasLimit;
+    emit GasLimitUpdated(gasLimit);
+  }
+
+  /**
+   * @notice fetches the gas cost of sending a cross chain message
+   */
+  function getGasLimit() external view returns (uint32) {
+    return s_gasLimit;
+  }
+
+  /**
+   * @notice makes this contract payable
+   * @dev receives funds:
+   *  - to use them (if configured) to pay for L2 execution on L1
+   *  - when withdrawing funds from L2 xDomain alias address (pay for L2 execution on L2)
+   */
+  receive() external payable {}
+
+  /**
+   * @notice withdraws all funds available in this contract to the msg.sender
+   * @dev only owner can call this
+   */
+  function withdrawFunds() external onlyOwner {
+    address payable recipient = payable(msg.sender);
+    uint256 amount = address(this).balance;
+    Address.sendValue(recipient, amount);
+  }
+
+  /**
+   * @notice withdraws all funds available in this contract to the address specified
+   * @dev only owner can call this
+   * @param recipient address where to send the funds
+   */
+  function withdrawFundsTo(address payable recipient) external onlyOwner {
+    uint256 amount = address(this).balance;
+    Address.sendValue(recipient, amount);
+  }
+
+  /**
+   * @notice withdraws funds from L2 xDomain alias address (representing this L1 contract)
+   * @dev only owner can call this
+   * @param amount of funds to withdraws
+   * @param refundAddr address where gas excess on L2 will be sent
+   *   WARNING: `refundAddr` is not aliased! Make sure you can recover the refunded funds on L2.
+   */
+  function withdrawFundsFromL2(uint256 amount, address refundAddr) external onlyOwner {
+    // Third parameter is _l1Gas and is unused in the function.  It is only included there for forward compatibility
+    // Fourth parameter is optional bytes
+    bytes memory message = abi.encodeWithSelector(
+      IL2ERC20Bridge.withdraw.selector,
+      s_L2ETHTokenAddr,
+      address(this),
+      0,
+      ""
+    );
+    IL1CrossDomainMessenger(L1_CROSS_DOMAIN_MESSENGER_ADDRESS).sendMessage(s_L2TokenBridgeAddr, message, s_gasLimit);
+    emit L2WithdrawalRequested(amount, refundAddr);
+  }
+
+  /**
+   * @notice sets config AccessControllerInterface contract
+   * @dev only owner can call this
+   * @param accessController new AccessControllerInterface contract address
+   */
+  function setConfigAC(address accessController) external onlyOwner {
+    _setConfigAC(accessController);
   }
 
   /**
    * @notice validate method sends an xDomain L2 tx to update Flags contract, in case of change from `previousAnswer`.
-   * @dev A message is sent via the Optimism CrossDomainMessenger L1 contract. The "payment" for L2 execution happens on L1,
-   *   using the gas attached to this tx (some extra gas is burned by the Optimism bridge to avoid DoS attacks).
-   *   This method is accessed controlled.
+   * @dev A retryable ticket is created on the Optimism L1 Inbox contract. The tx gas fee can be paid from this
```

Comment:
Does the optimism bridge refer to cross-chain messages as retryable tickets? We should use the terms they use.